asm/amd/regs_state.go

```diff
@@ -3,64 +3,310 @@
 
 package amd // import "go.opentelemetry.io/ebpf-profiler/asm/amd"
 
-import "golang.org/x/arch/x86/x86asm"
+import (
+	"errors"
+	"fmt"
+	"io"
+	"math"
+
+	"go.opentelemetry.io/ebpf-profiler/asm/variable"
+	"golang.org/x/arch/x86/x86asm"
+)
+
+const (
+	size64 = 64
+	size32 = 32
+	size16 = 16
+	size08 = 8
+)
+
+const debugPrinting = false
+
+type regIndexTableEntry struct {
+	idx int
+	sz  int
```

Comment:
nit: This apparently the bit size. Normally (at least to me) `sz` means bytes. Perhaps use `bits` here and all the other similar places?

Reply:
thanks, agree. renamed to `bits`